src/WebJobs.Script/Host/ScriptHostIdProvider.cs

```diff
@@ -26,6 +26,7 @@ public ScriptHostIdProvider(IConfiguration config, IEnvironment environment, IOp
             _config = config;
             _environment = environment;
             _options = options.Value;
+            _options.RootScriptPath = _options.RootScriptPath ?? config[EnvironmentSettingNames.AzureWebJobsScriptRoot];
```

Comment:
We should never update config anywhere. If the setup is missing a scenario, it needs to be updated to handle it.